portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.js

```diff
@@ -117,7 +116,7 @@ const postUtilisationReportUpload = async (req, res) => {
     }
 
     // File is valid so we can start processing and validating its data
-    const { csvJson, fileBuffer, error } = await extractCsvData(req.file); // do we here catch some errors and return generic something went wrong here?
+    const { csvJson, fileBuffer, error } = await extractCsvData(req.file);
```

Comment:
Do we need to have `extractCsvData` here? I think such file buffers should be handled by the API and the UI just consumes it.

Would like to avoid the UI having to do XYZ because the API doesn't return a desired/ideal response

Reply:
Totally agree with you on that, here we do multiple things with the extracted data, including attach it to the users session I believe so that we can use it on the next page since they have to confirm before uploading, so it is required to live here.

Reply:
I did briefly think about this when I started but discarded the idea but thinking about it more now I agree with where you are coming from. It is however a lot of additional work and having spoken to Christian there would be a few other things to consider:
- Increasing the number of file scans we do through clamscan
- Introducing multer and a bunch of context around file uploading to dtfs-central-api when the actual file gets uploaded to Azure in portal-api

If we were going to make this change it would need to be a follow on ticket, but would need to consider a bit more if it is something we should do.